### ISSUE TITLE
Custom emojis in the name is misaligned in iOS 15

### ISSUE THREAD

**Issue by eai04191** [metabolist/metatext#45]
**Describe the bug**
SSIA 
![image](https://user-images.githubusercontent.com/3516343/134128566-03aa8a65-421a-47b6-b68e-c339950fae7f.png)

**To Reproduce**
1. Update iOS to 15
2. Open a profile with a custom emojis in the name (eg [me](https://stellaria.network/@Eai))
3. See broken names on the timeline.

**Expected behavior**
Custom emojis and name letters should be in the same alignment

**Screenshots**
<details>
<summary>iOS 14</summary>

![image](https://user-images.githubusercontent.com/3516343/134128235-8c8ab99f-de51-4044-af05-0afc75f2ad74.png)
</details>

<details>
<summary>iOS 15</summary>

![image](https://user-images.githubusercontent.com/3516343/134128283-966f0530-24ac-4ce4-8e49-2cca815f4054.png)
</details>

 - Device: iPhone SE (2nd Gen) [MHGQ3J/A]
 - OS: iOS 15.0
 - App Version: 1.5.0(4)

**Comment by ShredderWalnut** [comment]
Same here.

- Device: iPhone X
- OS: iOS 15.0 (19A5340a)
- App Version: 15.0(4)

![IMG_4301](https://user-images.githubusercontent.com/34428332/134603245-73f30850-6042-45c3-892e-63aae87d35ec.jpg)

**Comment by genebean** [comment]
Same on iPhone 12 running iOS 15.1.1. I just installed the app and immediately had trouble reading toots because of this.

**Comment by greyivy** [comment]
This is becoming a major blocker to using this app on instances that rely on emotes to convey important information. As a non-iOS developer, is there any way I can help get this fixed?

**Comment by prplecake** [comment]
I think I've narrowed it down to this line: https://github.com/metabolist/metatext/blob/032e0040ed49f3d0a41ce5988394be23c1d64f0c/Extensions/NSMutableAttributedString%2BExtensions.swift#L41

Specifically the `y` value, I think... but I can't for the life of me figure out a better value, or have yet to find a value that aligns correctly.

**Comment by LeviSnoot** [comment]
Any news on this? Still experiencing the same on iOS 15.4.1, iPhone 13 Pro.

**Comment by mflint** [comment]
This _is_ a change of behaviour between iOS 14 and iOS 15 - the `imageBounds` parameter to `AnimatedTextAttachment.image(forBounds:textContainer:characterIndex)` is different for the two versions of iOS.

For iOS 14, `imageBounds` has a sensible-looking `y` value, close to zero. But on iOS 15 runtime, the `y` value is close to the height of the view. (20px or whatever).

I can't explain why ¯\\\_(ツ)_/¯

But I have done an ugly fix to `AnimatedAttachmentLabel`:

```
diff --git a/Views/UIKit/AnimatedAttachmentLabel.swift b/Views/UIKit/AnimatedAttachmentLabel.swift
index 91402d9..c2e30fb 100644
--- a/Views/UIKit/AnimatedAttachmentLabel.swift
+++ b/Views/UIKit/AnimatedAttachmentLabel.swift
@@ -28,7 +28,7 @@ final class AnimatedAttachmentLabel: UILabel, EmojiInsertable {
                   let imageBounds = animatedAttachment.imageBounds
             else { return }
 
-            animatedAttachment.imageView.frame = imageBounds
+            animatedAttachment.imageView.frame = CGRect(x: imageBounds.minX, y: (rect.height - imageBounds.height) / 2, width: imageBounds.width, height: imageBounds.height)
             animatedAttachment.imageView.contentMode = .scaleAspectFit
 
             if animatedAttachment.imageView.superview != self {
```

This change _ignores_ the y-position of the `imageBounds`, and instead centers the image vertically in the `AnimatedAttachmentLabel`.

(Note: this isn't a good long-term solution, because I think it will break multi-line labels)

**Comment by stramargio** [comment]
🙏🏻 @prplecake

**Comment by prplecake** [comment]
Confirmed fixed in 1.5.1!

**Comment by stramargio** [comment]
🎉

**Comment by eai04191** [comment]
fixed in 3e93ced674b2a11da11d6f2ec59fd09245cc6ad1 👍
